scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -521,11 +521,11 @@ trait CounterVerification extends Job {
   }
 }
 
-private[scalding] object FlowStepStrategies {
+private[scalding] case class FlowStepStrategies[A]() extends Semigroup[FlowStepStrategy[A]] {
```

Comment:
It's not side-effecting. Nothing happens in plus. It is just creating a new strategy. Looks correct to me.